observation/store_test.go

```diff
@@ -113,9 +113,9 @@ func TestStore_SaveAllExecError(t *testing.T) {
 		Convey("When dBConnection.Exec returns an error", func() {
 			results, err := store.SaveAll([]*observation.Observation{inputObservation})
 
-			Convey("Then no results and the expected error are returned", func() {
-				So(results, ShouldBeNil)
-				So(err, ShouldResemble, mockError)
+			Convey("Then an empty set of results and nil error are returned", func() {
+				So(len(results), ShouldEqual, 0)
+				So(err, ShouldBeNil)
```

Comment:
perhaps ensure the mocked ExecNeoFunc was called? `So(len(conn.ExecNeoCalls()), ShouldEqual, 1)`

Reply:
I have added this line 👍